emcc.py

```diff
@@ -620,15 +620,23 @@ def make_js_executable(script):
   cmd = config.JS_ENGINE
   if settings.WASM_BIGINT:
     cmd.append('--experimental-wasm-bigint')
+
+  # Write script file with .cjs extension to force Node to commonjs mode.
+  # This prevents build failures when any parent directory has
+  # package.json with {"type": "module"}. In addition, write a shell
+  # wrapper that calls node with this .cjs script, because autoconf
+  # does not expect its conftest binary to have an extension.
+  # See https://github.com/emscripten-core/emscripten/issues/17431
+  cjs_script = script + '.cjs'
```

Comment:
I think we only want to do this if the name of the output file does not have an extension already.  e.g. `conftest`.   If the user does `-o out.js` it would be a little odd to have `out.js` be shell script that runs `out.cjs`.

So I think maybe we need both old and new methods.  You can do `if not shared.suffix(script):` to check this.